include/utility/dbus_utility.hpp

```diff
@@ -58,6 +58,95 @@ inline types::MapperGetObject getObjectMap(const std::string& objectPath,
     return getObjectMap;
 }
 
+/**
+ * @brief An API to get object, services, and interfaces for an interface.
+ *
+ * The API returns the map of object, services and interfaces in the
+ * subtree that implement a certain interface. If no interfaces are provided
+ * then all the objects, services and interfaces under the subtree will
+ * be returned.
+ *
+ * Note: It will be caller's responsibility to check for empty map returned and
+ * generate appropriate error.
+ *
+ * @param[in] i_subTree - Sub tree path to search for an interface.
+ * @param[in] i_depth - Maximum depth of the tree to search.
+ * @param[in] i_interfaces - List of interfaces to search.
+ *
+ * @return - A map of object and its related services and interfaces, if
+ *           success. If failed, empty map.
+ */
+
+inline types::MapperGetSubTree
+    getObjectSubTree(const std::string& i_subTree, const int& i_depth,
+                     std::vector<std::string> i_interfaces)
+{
+    types::MapperGetSubTree l_subTreeMap;
+
+    try
+    {
+        auto bus = sdbusplus::bus::new_default();
```

Comment:
coding guideline is not followed for the variables.
And below as well.

Reply:
Done